examples/HistogramBitmap/HistogramBitmap.cpp

```diff
@@ -0,0 +1,234 @@
+			outFile.write((char*) &reservedBytes, sizeof(uint32_t));
+			outFile.write((char*) &pixelDataOffset, sizeof(uint32_t));
+		}
+	};
+
+
+	struct BmpInfoHeader {
+		uint32_t sizeOfThisHeader = 40;
+		int32_t width = 256; // in pixels
+		int32_t height = 256;
+		uint16_t numberOfColorPlanes = 1; // must be 1
+		uint16_t colorDepth = 24;
+		uint32_t compressionMethod = 0;
+		uint32_t rawBitmapDataSize = 0;
+		int32_t horizontalResolution = 3780; // in pixel per meter //do i even need this?
+		int32_t verticalResolution = 3780; // in pixel per meter
+		uint32_t colorTableEntries = 0;
+		uint32_t importantColors = 0;
+
+		BmpInfoHeader(uint16_t imageSize){
+			width = imageSize;
+			height = imageSize;
+		}
+		void writeToFile(std::ofstream& outFile){
+			//writing like this to avoid padding issues
+			outFile.write((char*) &sizeOfThisHeader, sizeof(uint32_t));
+			outFile.write((char*) &width, sizeof(int32_t));
+			outFile.write((char*) &height, sizeof(int32_t));
+			outFile.write((char*) &numberOfColorPlanes, sizeof(uint16_t));
+			outFile.write((char*) &colorDepth, sizeof(uint16_t));
+			outFile.write((char*) &compressionMethod, sizeof(uint32_t));
+			outFile.write((char*) &rawBitmapDataSize, sizeof(uint32_t));
+			outFile.write((char*) &horizontalResolution, sizeof(int32_t));
+			outFile.write((char*) &verticalResolution, sizeof(int32_t));
+			outFile.write((char*) &colorTableEntries, sizeof(uint32_t));
+			outFile.write((char*) &importantColors, sizeof(uint32_t));
+		}
+	};
+
+
+namespace {
+template <typename Type>
+concept NumberT = std::integral<Type> || std::floating_point<Type>;
+
+template <NumberT Type>
+auto parse_as(Type& out, std::string_view const value) -> bool {
+	if (value.empty()) { return false; }
+
+	auto const* end = value.data() + value.size();
+	auto const [ptr, ec] = std::from_chars(value.data(), end, out);
+
+	return ec == std::errc{} && ptr == end;
+}
+
+struct Args {
+	std::span<char const* const> args{};
+
+	[[nodiscard]] constexpr auto next() -> std::string_view {
+		if (args.empty()) { return {}; }
+		auto const* ret = args.front();
+		args = args.subspan(1);
+		return ret;
+	}
+
+	template <NumberT Type>
+	auto next_as(Type& out, std::string_view const key) -> bool {
+		auto value = next();
+		if (value.empty()) { return true; }
+
+		if (!parse_as(out, value)) {
+			std::cerr << std::format("invalid argument: '{}' for '{}'\n", value, key);
+			return false;
+		}
+
+		return true;
+	}
+};
+
+struct Config {
+	noiz::Seed seed{noiz::detail::Generator::make_random_seed()};
+	noiz::GridExtent2 grid_extent{50, 1}; // NOLINT
+	int count{100};						  // NOLINT
+	float step{0.1f};					  // NOLINT
+	int imageSize{1};
+
+	// syntax: [count] [step]
+	auto parse_args(Args args) -> bool {
+		if (!args.next_as(count, "count")) { return false; }
+		if (!args.next_as(step, "step")) { return false; }
+		if (!args.next_as(imageSize, "imageSize")) { return false; }
+		assert(imageSize > 0 && imageSize <= 16);
```

Comment:
Don't `assert` about an input that could be anything, it will unnecessarily crash the program. Instead print the error and return false. Similarly, the "unrecognized argument" part is handled here, so it doesn't need to be printed again in main.

```suggestion
		if (imageSize < 0 || imageSize > 16) {
		    std::cerr << std::format("invalid image size: '{}'\n", imageSize);
		    return false;
		}
```